runtime/j9vm/javanextvmi.cpp

```diff
@@ -272,16 +273,27 @@ exitVThreadTransitionCritical(J9VMThread *currentThread, j9object_t vthread)
 	J9OBJECT_I64_STORE(currentThread, vthread, currentThread->javaVM->virtualThreadInspectorCountOffset, 0);
 }
 
-JNIEXPORT void JNICALL
-JVM_VirtualThreadMountBegin(JNIEnv *env, jobject thread, jboolean firstMount)
+static void
+unsetParentVthread(J9VMThread *currentThread, jobject thread)
+{
+	enterVThreadTransitionCritical(currentThread, thread);
+	/* Re-fetch reference as enterVThreadTransitionCritical may release VMAccess. */
+	j9object_t threadObj = J9_JNI_UNWRAP_REFERENCE(thread);
+	j9object_t continuationObj = J9VMJAVALANGVIRTUALTHREAD_CONT(currentThread, threadObj);
+	/* Add reverse link from Continuation object to VirtualThread object, this let JVMTI code. */
+	J9VMJDKINTERNALVMCONTINUATION_SET_VTHREAD(currentThread, continuationObj, NULL);
+	exitVThreadTransitionCritical(currentThread, threadObj);
+}
+
+/* Caller must have VMAccess. */
+static void
+virtualThreadMountBegin(JNIEnv *env, jobject thread)
 {
 	J9VMThread *currentThread = (J9VMThread *)env;
 	J9JavaVM *vm = currentThread->javaVM;
```

Comment:
`vm` is only used within the `if` statement below, this declaration should be moved there.